File: api-tabung/src/main/java/com/izan/tabungapi/entity/Tabung.java
package com.izan.tabungapi.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Data
@Entity
@Table(name = "tabung")
public class Tabung {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @Column(name = "iscents")
    private boolean isCents;

    @ManyToOne
    @JoinColumn(name = "tabung_types_id")
    private TabungType tabungType;
}

File: api/tabung-api-go/model/tabung.go
package model

type Tabung struct {
	Id           int64      `gorm:"column:id;primaryKey" json:"id"`
	Name         string     `gorm:"column:name" json:"name"`
	IsCents      bool       `gorm:"column:is_cents" json:"cents"`
	TabungTypeId int64      `gorm:"column:tabung_types_id" json:"-"`
	TabungType   TabungType `gorm:"foreignKey:TabungTypeId" json:"tabungType"`
}

func (Tabung) TableName() string {
	return "tabung"
}
